cdap-client-tests/src/test/java/co/cask/cdap/client/WorkflowClientTestRun.java

```diff
@@ -76,10 +78,18 @@ public void testWorkflowClient() throws Exception {
     String outputPath = new File(tmpFolder.newFolder(), "output").getAbsolutePath();
     Map<String, String> runtimeArgs = ImmutableMap.of("inputPath", createInput("input"),
                                                       "outputPath", outputPath);
-    Id.Workflow workflowId = Id.Workflow.from(appId, AppWithWorkflow.SampleWorkflow.NAME);
+    final Id.Workflow workflowId = Id.Workflow.from(appId, AppWithWorkflow.SampleWorkflow.NAME);
     programClient.start(workflowId, false, runtimeArgs);
     programClient.waitForStatus(workflowId, "STOPPED", 60, TimeUnit.SECONDS);
 
+    Tasks.waitFor(1, new Callable<Integer>() {
+      @Override
+      public Integer call() throws Exception {
+        return programClient.getProgramRuns(workflowId,
+                                            ProgramRunStatus.COMPLETED.name(), 0, Long.MAX_VALUE, 10).size();
+      }
+    }, 10, TimeUnit.SECONDS, 100, TimeUnit.MILLISECONDS);
+
     List<RunRecord> workflowRuns = programClient.getProgramRuns(workflowId, ProgramRunStatus.COMPLETED.name(), 0,
                                                                 Long.MAX_VALUE, 10);
     Assert.assertEquals(1, workflowRuns.size());
```

Comment:
This assertion is no longer needed then, right? because it tests the same thing as `Task.waitFor` is doing.